File: Sources/iStream/Models/Field.swift
import Foundation

public enum FieldType: String, CaseIterable, Codable, Sendable {
    case string
    case int
    case long
    case double
    case decimal
    case timestamp
    case boolean
    case bytes
}

public struct Field: Identifiable, Hashable, Codable, Sendable {
    public var id: UUID
    public var name: String
    public var type: FieldType

    public init(id: UUID = UUID(), name: String, type: FieldType) {
        self.id = id
        self.name = name
        self.type = type
    }
}

File: lib/models/field.dart
enum FieldType {
  stringType('string'),
  intType('int'),
  longType('long'),
  doubleType('double'),
  decimalType('decimal'),
  timestampType('timestamp'),
  booleanType('boolean'),
  bytesType('bytes');

  const FieldType(this.label);
  final String label;
}

class Field {
  Field({required this.name, required this.type});

  String name;
  FieldType type;
}
